execution/src/posix/execution.cc

```diff
@@ -463,11 +448,26 @@ namespace leatherman { namespace execution {
                                             options[execution_options::inherit_locale]);
         auto envp = to_exec_arg(&variables);
 
+        // Set all open file descriptors above stderr to close on exec
+        auto max_desc_limit = get_max_descriptor_limit();
```

Comment:
Isn't this a timing issue where one thread marked the descriptors close-on-exec just prior to the `vfork`, another thread just completed a `vfork` and is now marking the same descriptors as not close-on-exec just in time for the first thread to `exec` without them flagged? 

Solaris is the worst, but Linux stubbornly not providing a `closefrom` equivalent is equally maddening (no really, we only want three open descriptors for our child process!).